### PR TITLE
Fixes testing web signups

### PR DESCRIPTION
**Pull request by aaronschachter**
Adds basic auth to a Content API request to fix the list of active campaigns, and replaces its command `keyword` to `web`, as Gambit Admin is the home for viewing current keywords -- but we still need ability to test web signups via Slack. 

Updates a lot of these functions for newer features:

* Deprecates the `thor`, `qa`, `staging` commands they are relics from the deprecated Contentful `keyword` content type

* Consolidates the `lib/gambit` dir into a single `lib/gambit.js` that integrates with the Gambit Admin API (which proxies requests to Campaigns and Conversations)

Part 2 cleanup in a future PR will entail:
* Moving `app/controllers/slack.js` into `lib/slack` to simplify dir structure, complete TODO of adding a `config/lib/slack.js`
* Moving cache into a helper lib

### CODE REVIEW

**Comment by aaronschachter** [comment]
@rapala61 This PR is not urgent for review, I've been testing this branch against our Gambit Staging app.
 
I don't think we need to worry about this -- but make known a gotcha I've found when testing web signups over Slack when testing a campaign I'm not signed up for, and then continuing the conversation.  If I send a Slack reply to the web signup test sent, I get a signup confirmation sent to my mobile number, because my Slack message in the topic creates a signup (since I'm currently in a topic for a campaign and a signup doesn't exist. A `POST messages?origin=signup` Conversation request doesn't make any Rogue requests, it just sets the user's conversation topic accordingly). 

cc @ngjo

File: app/controllers/slack.js [hunk 1]
```diff
@@ -50,29 +48,25 @@ function fetchNorthstarUserForSlackUserId(slackUserId) {
...
-    .then((response) => {
-      const activeCampaigns = response.body.data.filter(campaign => campaign.status === 'active');
-      const text = `Active campaigns on Gambit ${environmentName.toUpperCase()}:`;
-      const attachments = activeCampaigns.map((campaign, index) => {
-        const attachment = slack.parseCampaignAsAttachment(environmentName, campaign, index);
-        return attachment;
-      });
-
-      return web.chat.postMessage(channel, text, { attachments });
+  return gambit.get('campaigns')
```
Comment (rapala61) [inline app/controllers/slack.js]: Nice naming change. It made so much simple to reason. 👍

File: app/controllers/slack.js [hunk 2]
```diff
@@ -148,32 +124,21 @@ rtm.on(Slack.RTM_EVENTS.MESSAGE, (message) => {
...
-    return postCampaignIndexMessage(channel, 'thor');
+  if (command === 'web') {
+    return sendWebSignupList(channel);
   }
+
   if (command === 'broadcast') {
     const broadcastId = message.broadcastId;
     if (!broadcastId) {
-      const errorMsg = 'You need to pass a Broadcast Id. Example:\n\n> broadcast 5mPrrJjImQAGYi4goYWk2S`';
+      const errorMsg = 'You need to pass a Broadcast Id. Example:\n\n> broadcast 2en018uiWcsMcIAWsGCQwS`';
```
Comment (rapala61) [inline app/controllers/slack.js]: Is that an extra `?

Comment (aaronschachter) [reply app/controllers/slack.js]: 🦅  👀 🍻  -- found this after jumping into the deep end and creating a separate branch for a part 2 in #42

**Pull request merged**